source/elements/oneMKL/source/domains/spblas/data_types/set_matrix_property.rst

```diff
@@ -0,0 +1,95 @@
+.. SPDX-FileCopyrightText: 2024 Intel Corporation
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+.. _onemkl_sparse_set_matrix_property:
+
+set_matrix_property
+===================
+
+Matrix properties
+-----------------
+
+.. container:: section
+
+   .. code:: cpp
+
+      enum class matrix_property {
+          symmetric,
+          sorted,
+      };
+
+   TODO: Clarify are the properties meant to be used as a bit mask so multiple
+   properties can be set?
+
+   .. list-table::
+      :header-rows: 1
+      :widths: 20 80
+
+      * - Value
+        - Description
+      * - ``symmetric``
+        - Guarantees that the matrix is symmetric, meaning it must be square and
+          its transpose is equal to itself. TODO: Confirm whether the user can
+          only provide half of the data in this case?
+      * - ``sorted``
+        - Guarantees that the column indices are sorted for a given row. Only
+          available for the CSR format. TODO: Confirm the description.
```

Comment:
From what I understand it is the column indices within each rows that are sorted (i.e. the `col_ind` parameter). So potentially `col_sorted` would be a better name? It's not shown here but Gajanan sent me some potential future properties which could be `sorted_by_row` for the CSR, COO and BSR formats and `sorted_by_column` for the CSC and COO formats.

I don't get how `sorted_by_row` would be different from `sorted` for CSR in this suggestion? We could maybe already use ` sorted_by_rows` and `sorted_by_columns` properties and not have `sorted`?

The intention was to silently ignore the properties if they are not supported. I didn't think `sorted` would make sense for COO format but you're right.

Reply:
* For CSR, *sorted_by_row* would be equivalent to *sorted* for the CSR. However, if one were to apply the given description to other matrix formats (ie. COO), this wouldn't be true.
  * I worry that I'm making something that isn't a problem into a problem. If this fits with other sparse APIs you've seen, I think `sorted` is fine.
* I agree that `row_sorted` meaning that the elements in the row are sorted or that the rows themselves are sorted would cause confusion, and prefer your suggestions.

Reply:
I think this is a good question that could be clarified. There isn't a good equivalent for these matrix properties in other libraries, only MKL backend would use them currently. cuSPARSE for instance "only" documents which parameters are can be sorted and which cannot. Similarly, the COO format can be unsorted, sorted by rows or sorted by columns whereas CSR can only be unsorted or sorted which makes sense.

Currently my suggestions would either be to keep the `sorted` property as it is or replace it with `sorted_by_columns` for COO and CSR format and `sorted_by_rows` for COO format. I have a small preference for the latter. @gajanan-choudhary's input would be useful here.

Reply:
We discussed this separately. We decided to keep the `sorted` property and make sure it has a meaning for each format. The meaning can vary for each format. This was reworded in https://github.com/oneapi-src/oneAPI-spec/pull/522/commits/b4c2a467f235d21df79e7efd8216e665d4357fdf